Scripts/Core/BrushBase.cs

```diff
@@ -10,6 +10,7 @@
 namespace Sabresaurus.SabreCSG
 {
 	public enum CSGMode { Add, Subtract };
+    [ExecuteInEditMode]
```

Comment:
Have you tested this against the test scene with a thousand brushes in, might be worth checking against levels with high brush counts if this dramatically increases scene load times or slows down interactivity

Reply:
I have tested it against Temple several times with and without that statement and it makes no difference to the tool performance and loading time of the scene in the editor (~ 9 seconds).

PrimitiveBrush, GroupBrush, StairBrush, TrimBrush, CurvedStairBrush and ShapeEditorBrush all have [ExecuteInEditMode]. Essentially all of them except the abstract CompoundBrush. I went ahead and cleared my logic from BaseBrush.Update() as it's no longer necessary as well.

All it does now is a quick name comparison when the scene loads or when a brush is created.